src/tests/mutation_log_learn.cpp

```diff
@@ -119,10 +126,22 @@ TEST(replication, log_learn)
                     );
                 EXPECT_TRUE(wmu->client_requests.size() == mu->client_requests.size());
                 EXPECT_TRUE(wmu->client_requests[0].code == mu->client_requests[0].code);
+
+                std::lock_guard<std::mutex> _(lock);
+                replay_done = true;
+                cv.notify_all();
                 return true;
             },
             offset
             );
+        //wait for replay done
+        {
```

Comment:
mutation_log::replay is a sync call, why do you need to do synchronization here?